pkg/render/intrusiondetection/dpi/dpi_test.go

```diff
@@ -243,6 +268,47 @@ var _ = Describe("DPI rendering tests", func() {
 		validateDPIComponents(resources, false)
 	})
 
+	It("should render all resources for deep packet inspection with default resource requirements for a management cluster without DPI resources", func() {
+		cfg = &dpi.DPIConfig{
+			IntrusionDetection: ids,
+			Installation:       installation,
+			TyphaNodeTLS:       typhaNodeTLS,
+			PullSecrets:        pullSecrets,
+			Openshift:          false,
+			HasNoLicense:       false,
+			HasNoDPIResource:   true,
+			ManagementCluster:  true,
+			ESClusterConfig:    esConfigMap,
+			ClusterDomain:      dns.DefaultClusterDomain,
+			DPICertSecret:      dpiCertSecret,
+		}
+		component := dpi.DPI(cfg)
+
+		resources, _ := component.Objects()
+
+		expectedResources := []resourceTestObj{
+			{name: dpi.DeepPacketInspectionNamespace, ns: "", group: "", version: "v1", kind: "Namespace"},
+			{name: dpi.DeepPacketInspectionLinseedRBACName, ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
+			{name: dpi.DeepPacketInspectionLinseedRBACName, ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
+		}
+
+		Expect(len(resources)).To(Equal(len(expectedResources)))
+
+		for i, expectedRes := range expectedResources {
+			rtest.CompareResource(resources[i], expectedRes.name, expectedRes.ns, expectedRes.group, expectedRes.version, expectedRes.kind)
```

Comment:
This is a better function to use in tests: https://github.com/tigera/operator/blob/0c1c706b30dc8ab66247bdc911d005a71b4283ee/pkg/render/common/test/testing.go#L107-L138

I know this is just copied from the other tests in this package, but for new tests we should be using this function IMO - it produces much better output and is more flexible to changes in ordering, etc.

Reply:
If you want 100% coverage, this function is even better: https://github.com/tigera/operator/blob/0c1c706b30dc8ab66247bdc911d005a71b4283ee/pkg/render/common/test/testing.go#L95